src/app/donation-start/donation-start.component.ts

```diff
@@ -239,6 +239,17 @@ export class DonationStartComponent implements AfterContentChecked, OnDestroy, O
   }
 
   async stepChanged(event: StepperSelectionEvent) {
+
+    // If the original donation amount was updated, cancel that donation,
+    // we'll create a new one later on for this updated amount.
+    if (this.donation !== undefined && this.donationAmount !== this.donation.donationAmount) {
+      this.analyticsService.logEvent(
+        'cancel_auto',
+        `Donation cancelled because amount was updated ${this.donation.donationId} to campaign ${this.campaignId}`,
+      );
+      this.donationService.cancel(this.donation).subscribe(() => this.donationService.removeLocalDonation(this.donation));
```

Comment:
What about `this.donation`?

Reply:
@NoelLH I don't think I follow, the if check ensures the form amount is different to the object amount, and it's then cancelling the existing donation

Reply:
@NoelLH Doing a delete on `this.donation` when the amount changes causes the offer existing donation dialog box to not appear on refresh, I've tested the current approach and it works for all scenarios

Reply:
@tbg-jasonfung yes, as per your edit, they can't reuse a cancelled donation. So shouldn't we put the local var deletion in?

Reply:
@NoelLH Once the amount has changed and the existing donation is cancelled and a new donation is created, even then on refresh the existing donation dialog box doesn't appear :/

Reply:
@tbg-jasonfung that can't be to do with the removal of `this.donation`, which is only in local state, since that doesn't impact the new page on refresh. So I don't understand why we're not removing it? It sounds like there may also be a separate issue either way in the case you describe.

Reply:
@NoelLH Think i figured it out, it was deleting the local donation before `removeLocalDonation` was being triggered, so on refresh it found the donation that was `Cancelled` but there's a method that says, don't offer this as a previous donation if the status is not `Pending` or  `Reserved`